ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/FieldValidations.scala

```diff
@@ -99,32 +102,64 @@ trait FieldValidations {
   def requirePresence[T](option: Option[T], fieldName: String): Either[StatusRuntimeException, T] =
     option.fold[Either[StatusRuntimeException, T]](Left(missingField(fieldName)))(Right(_))
 
-  def validateDeduplicationDuration(
-      durationO: Option[com.google.protobuf.duration.Duration],
+  /** We validate only using current time because we set the currentTime as submitTime so no need to check both
+    */
+  def validateDeduplicationPeriod(
```

Comment:
Do these checks run on all submissions or only for those where the WriteService has https://github.com/digital-asset/daml/blob/main/ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/WriteService.scala#L110 set to `true`? I'd strongly prefer it this kind of validation is done by the WriteService that's responsible for command deduplication. Otherwise, we're again using different clocks at different points on the submission path, which is a recipe for sporadic failures..

Reply:
@andreaslochbihler-da 
Currently it runs for all submissions. I agree that it would a lot better to just validate for api deduplication when the api deduplication is enabled. i could for sure change that. Do you think it would make more sense to remove the validation all together?
@SamirTalwar-DA Any thoughts on this?

Reply:
I don't know. The validation may be useful for other implementations of the read/write service. But then this should be a utility function that lives somewhere where the `DeduplicationPeriod`s are defined, shouldn't it?

Reply:
Sorry, I missed this question.

I agree that this should be done by the driver (on the participant or the committer, as necessary), not the ledger API server, with the same clock and definition of "now" as is going to be used for the submission itself.

Reply:
As we removed for now the deduplication period which specifies a timestampt, and for offset we currently have no way of validating here, we only validate when receiving a duration so this is no longer relevant.